### PR TITLE
Fix Warnings

### PR DESCRIPTION
**Pull request by donpdonp**
### Description
Fixes compiler warnings. Changed unused struct fields from name to _name. Moves 'code-cov' and 'generic.bitcoin-tests' dockerfiles from nightly to stable.

### Applicable issues
- fixes #

### CODE REVIEW

File: testnet/stacks-node/src/burnchains/bitcoin_regtest_controller.rs [hunk 1]
```diff
@@ -1660,10 +1660,10 @@ pub struct ParsedUTXO {
     script_pub_key: String,
     amount: Box<RawValue>,
     confirmations: u32,
-    spendable: bool,
-    solvable: bool,
-    desc: Option<String>,
-    safe: bool,
+    _spendable: bool,
```
[the remark this answers is not in the document]
Comment (donpdonp) [reply testnet/stacks-node/src/burnchains/bitcoin_regtest_controller.rs:1663]: My intuition was the struct represents something from a bitcoin rpc response, so having the fields in there is a useful reference to whats available. Looking further, I think its something for internal use after all, so its a judgement call whether to keep them or not. Looking at the previous commit the lines were added by @lgalabru so perhaps he has an opinion.

**Comment by jcnelson** [comment]
Looks like we got some compile errors.

**Comment by donpdonp** [comment]
On my laptop with the same commit hashes as the code-cov report "Merging #3298 (acc120f) into master (3e8a917)", cargo build completes with no warnings or errors. So I'm trying to figure out how the setup for this PR is leading to those std::fs errors.

**Comment by donpdonp** [comment]
The warnings and errors are cleaned up - the difference was due to rust 1.64 in CI and 1.63 on my laptop. I thought about changing rust-toolchain to "1.64" but a) the changes also work in 1.63  b) there isn't a way to say 1.64 "and higher", so it remains at 'stable'. 

There were two checks called build-publish-<something> and they failed for want of a docker hub username and password. Whats the fix for those steps?

**Comment by donpdonp** [comment]
closing this and re-creating for `next` branch

**Pull request closed without merging**